velox/exec/HashProbe.cpp

```diff
@@ -1145,13 +1145,32 @@ int32_t HashProbe::evalFilter(int32_t numRows) {
   if (!filter_) {
     return numRows;
   }
+
   const bool filterPropagateNulls = filter_->expr(0)->propagatesNulls();
   auto* rawOutputProbeRowMapping =
       outputRowMapping_->asMutable<vector_size_t>();
 
-  fillFilterInput(numRows);
   filterInputRows_.resizeFill(numRows);
 
+  // For anti join, do not evaluate filter on rows without a match. Evaluating
+  // filter on rows without a match may trigger errors in filter evaluation and
+  // fail the query unnecessarily.
+  // TODO Apply to the same to left joins.
+  if (isAntiJoin(joinType_)) {
```

Comment:
Should we do the same to null-aware anti join?

Reply:
That's a good question. I have an AI to look into that.